src/test/unit/ArgsParserTest.groovy

```diff
@@ -0,0 +1,252 @@
+package unit
+
+import arguments.ArgsParser
+import arguments.Arguments
+import exception.InvalidArgsException
+import org.junit.Test
+
+import static util.Assert.*
+
+public class ArgsParserTest {
+
+    ArgsParser argsParser = new ArgsParser()
+
+    @Test
+    public void withManyArgument() {
+        String[] args = ['-a', 'token',
+                         '-i', 'injectors.StaticAnalysisConflictsDetectionModule',
+                         '-k',
+                         '-p', 'https://github.com/mock-project',
+                         '-s', '01/01/2000',
+                         '-u', '01/01/2000',
+                         '-t', '2',
+                         './projects.csv', 'SOOTAnalysisOutput']
+        Arguments resultArgs = argsParser.parse(args)
+
+        assertEquals(resultArgs.accessKey,  'token')
+        assertEquals(resultArgs.isHelp(),  false)
+        assertEquals(resultArgs.injector.toString(),  'class injectors.StaticAnalysisConflictsDetectionModule')
+        assertEquals(resultArgs.keepProjects,  true)
+        assertEquals(resultArgs.resultsRemoteRepositoryURL,  'https://github.com/mock-project')
+        assertEquals(resultArgs.sinceDate,  '01/01/2000')
+        assertEquals(resultArgs.numOfThreads,  2)
+        assertEquals(resultArgs.untilDate,  '01/01/2000')
+        assertEquals(resultArgs.inputPath,  './projects.csv')
+        assertEquals(resultArgs.outputPath,  'SOOTAnalysisOutput')
+    }
+
+    @Test
+    public void defaultArgument() {
+        String[] args = ['./projects.csv',]
+        Arguments resultArgs = argsParser.parse(args)
+
+        assertEquals(resultArgs.accessKey,  '')
+        assertEquals(resultArgs.isHelp(),  false)
+        assertEquals(resultArgs.injector.toString(),  'class injectors.StaticAnalysisConflictsDetectionModule')
+        assertEquals(resultArgs.keepProjects,  false)
+        assertEquals(resultArgs.resultsRemoteRepositoryURL,  '')
+        assertEquals(resultArgs.sinceDate,  '')
+        assertEquals(resultArgs.numOfThreads,  1)
+        assertEquals(resultArgs.untilDate,  '')
+        assertEquals(resultArgs.inputPath,  './projects.csv')
+        assertEquals(resultArgs.outputPath,  'output')
+    }
+
+    @Test
+    public void onlyHelpArgument() {
+        String[] help = ['--help']
+        Arguments resultArgsHelp = argsParser.parse(help)
+
+        assertEquals(resultArgsHelp.isHelp(),  true)
+
+        String[] argh = ['-h']
+        Arguments resultArgsH = argsParser.parse(argh)
+
+        assertEquals(resultArgsH.isHelp(),  true)
+    }
+
+    @Test
+    public void withoutArgs() {
+        String[] args = []
+        Arguments resultArgs = argsParser.parse(args)
+
+        Arguments argsDefault = new Arguments()
+        argsDefault.setHelp()
+
+        assertEquals(argsDefault.getProperties(), resultArgs.getProperties())
+    }
+
+    @Test
+    public void invalidArgument() {
```

Comment:
Is there any way we could improve this behavior? throwing an error that better represents this scenario, something like: "Invalid argument passed"

Reply:
I believe that if we place a new validation verifying if the arguments passed are different from those expected, we would throw an exception with a personalized message. Should I do this?

Reply:
yes, I think this would be a good fix

Reply:
correction made on: https://github.com/spgroup/miningframework/pull/96/commits/410c4d2e9bc77ddb9fbf22373edc2e6c07c02cd6